lib/reactor/builder.ex

```diff
@@ -256,16 +289,33 @@ defmodule Reactor.Builder do
     end)
   end
 
+  defp maybe_build_step_transform_step(arguments, _name, nil), do: {:ok, arguments, nil}
+
+  defp maybe_build_step_transform_step(arguments, name, transform)
+       when is_function(transform, 1),
+       do: maybe_build_step_transform_step(arguments, name, {Step.TransformAll, fun: transform})
+
+  defp maybe_build_step_transform_step(arguments, name, transform) do
+    step = %Step{
+      arguments: arguments,
+      async?: true,
+      impl: transform,
+      name: {:transform, :for, name},
```

Comment:
I think these builtin step names might potentially need some "identifier that people would never use" that we can document should not be used. i.e `{:__reactor__, ...}`